distributed/tests/test_client.py

```diff
@@ -2710,29 +2710,78 @@ def test_run_coroutine_sync(c, s, a, b):
     assert t2 - t1 <= 1.0
 
 
-@gen_cluster(client=True)
-async def test_run_exception(c, s, a, b):
+@pytest.mark.parametrize(
+    "Worker,nanny,address_attr,log_name",
+    [
+        (Worker, False, "address", "worker"),
+        (Nanny, True, "worker_address", "nanny"),
+    ],
+)
+@gen_cluster(client=True, nthreads=[])
+async def test_run_exception(c, s, Worker, nanny, address_attr, log_name):
```

Comment:
nit: This test has become quite a handful to read. Would it make sense to split this up into smaller and more descriptive parts?